### PR TITLE
feat(azure): Add new check `storage_key_rotation_90_days`

### PR DESCRIPTION
**Pull request by pedrooot**
### Context

Add new check for Azure provider in Prowler "storage_key_rotation_90_days"

### Description

This check makes this task: 
* Ensure that 'Enable key rotation reminders' is enabled for each Storage Account

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days.metadata.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  "CheckID": "storage_key_rotation_90_days",
+  "CheckTitle": "Ensure that 'Enable key rotation reminders' is enabled for each Storage Account",
+  "CheckType": [],
+  "ServiceName": "storage",
+  "SubServiceName": "",
+  "ResourceIdTemplate": "",
+  "Severity": "medium",
+  "ResourceType": "AzureStorageAccount",
+  "Description": "Ensure that 'Enable key rotation reminders' is enabled for each Storage Account",
+  "Risk": "TLS versions 1.0 and 1.1 are known to be susceptible to certain Common Vulnerabilities and Exposures (CVE) weaknesses and attacks such as POODLE and BEAST",
```
Comment (jfagoagas) [inline prowler/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days.metadata.json]: I think this is not for this check, please review all metadata fields.

File: prowler/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days.metadata.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  "RelatedUrl": "",
```
Comment (jfagoagas) [inline prowler/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days.metadata.json]:
```suggestion
  "RelatedUrl": "https://learn.microsoft.com/en-us/azure/storage/common/storage-account-keys-manage?tabs=azure-portal",
```

File: prowler/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days.metadata.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  "Remediation": {
+    "Code": {
+      "CLI": "https://learn.microsoft.com/en-us/azure/storage/common/storage-account-keys-manage?tabs=azure-portal",
+      "NativeIaC": "",
+      "Other": "https://www.tenable.com/audits/items/CIS_Microsoft_Azure_Foundations_v1.5.0_L1.audit:5c2be112f94eebb430bfd55752740ab4",
+      "Terraform": ""
+    },
+    "Recommendation": {
+      "Text": "Ensure that all your Microsoft Azure Storage accounts are using the latest available version of the TLS protocol.",
```
Comment (jfagoagas) [inline prowler/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days.metadata.json]:
```suggestion
      "Text": "Ensure that Azure Storage account access keys are regenerated every 90 days in order to decrease the likelihood of accidental exposures and protect your storage account resources against unauthorized access.",
```

File: prowler/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+class storage_key_rotation_90_days(Check):
+    def execute(self) -> Check_Report_Azure:
+        findings = []
+        for subscription, storage_accounts in storage_client.storage_accounts.items():
+            for storage_account in storage_accounts:
+                report = Check_Report_Azure(self.metadata())
+                report.subscription = subscription
+                report.status = "PASS"
+                report.status_extended = f"Storage account {storage_account.name} from subscription {subscription} has a valid key expiration period of {storage_account.key_expiration_period_in_days} days."
```
Comment (jfagoagas) [inline prowler/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days.py]: Maybe we can get rid of the `valid` word, what do you think?
```suggestion
                report.status_extended = f"Storage account {storage_account.name} from subscription {subscription} has a key expiration period of {storage_account.key_expiration_period_in_days} days."
```

File: tests/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days_test.py [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+        ):
+            from prowler.providers.azure.services.storage.storage_key_rotation_90_days.storage_key_rotation_90_days import (
+                storage_key_rotation_90_days,
+            )
+
+            check = storage_key_rotation_90_days()
+            result = check.execute()
+            assert len(result) == 0
+
+    def test_storage_storage_key_rotation_91_days(self):
```
Comment (jfagoagas) [inline tests/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days_test.py:25]: Why more than 90 days is a FAIL? Just to understand.

**Review by jfagoagas — DISMISSED** [review]
Great one @pedrooot 👏  Please be careful when filling the metadata file, we have to always fill the remediation, recommendation, description and links, check that.

Let's review this comments with @sergargar to finish the PR and merge it.

File: tests/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days_test.py (hunk 5 above)
Comment (pedrooot) [reply tests/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days_test.py:25]: I took the recommendation from https://www.trendmicro.com/cloudoneconformity-staging/knowledge-base/azure/StorageAccounts/regenerate-storage-account-access-keys-periodically.html#

File: prowler/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days.py (hunk 4 above)
Comment (pedrooot) [reply prowler/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days.py]: Nice! Done

File: prowler/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days.metadata.json (hunk 3 above)
Comment (pedrooot) [reply prowler/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days.metadata.json]: Done!

File: prowler/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days.metadata.json (hunk 2 above)
Comment (pedrooot) [reply prowler/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days.metadata.json]: Done

File: prowler/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days.metadata.json (hunk 1 above)
Comment (pedrooot) [reply prowler/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days.metadata.json]: Done!

File: prowler/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        findings = []
+        for subscription, storage_accounts in storage_client.storage_accounts.items():
+            for storage_account in storage_accounts:
+                report = Check_Report_Azure(self.metadata())
+                report.subscription = subscription
+                report.status = "PASS"
+                report.status_extended = f"Storage account {storage_account.name} from subscription {subscription} has a key expiration period of {storage_account.key_expiration_period_in_days} days."
+                report.resource_name = storage_account.name
+                report.resource_id = storage_account.id
+                if storage_account.key_expiration_period_in_days is None:
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days.py]:
```suggestion
                if not storage_account.key_expiration_period_in_days:
```

File: prowler/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+class storage_key_rotation_90_days(Check):
+    def execute(self) -> Check_Report_Azure:
+        findings = []
+        for subscription, storage_accounts in storage_client.storage_accounts.items():
+            for storage_account in storage_accounts:
+                report = Check_Report_Azure(self.metadata())
+                report.subscription = subscription
+                report.status = "PASS"
+                report.status_extended = f"Storage account {storage_account.name} from subscription {subscription} has a key expiration period of {storage_account.key_expiration_period_in_days} days."
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/storage/storage_key_rotation_90_days/storage_key_rotation_90_days.py]: You are assigning `storage_account.key_expiration_period_in_days` to the string without checking if it exists first.

**Review by MrCloudSec — DISMISSED** [review]
Great work! 🚀

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**